src/typ.rs

```diff
@@ -2290,6 +2310,96 @@ impl<'a> Typer<'a> {
         })
     }
 
+    fn infer_record_update(
+        &mut self,
+        constructor: UntypedExpr,
+        spread: RecordUpdateSpread,
+        args: Vec<UntypedRecordUpdateArg>,
+        location: SrcSpan,
+    ) -> Result<TypedExpr, Error> {
+        let constructor = self.infer(constructor)?;
+
+        let (module, name) = match constructor {
```

Comment:
This block here could in theory be extracted out to a helper. It's very similar to `get_field_map`